contrib/cirrus/cache_release.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -e
+
+source $(dirname $0)/lib.sh
+
+req_env_var GOSRC CIRRUS_BUILD_ID
+
+cd $GOSRC
+
+make release_info.txt
+FIELDS="$RELEASE_BASENAME $RELEASE_VERSION $RELEASE_DIST $RELEASE_DIST_VER $RELEASE_ARCH $RELEASE_FILE"
+cat release_info.txt | read "$FIELDS"
+req_env_var $FIELDS
+
+if [[ "$CI" == "true" ]]
+then
+    req_env_var CIRRUS_BRANCH CIRRUS_TASK_NAME CIRRUS_PR
+
+    # Clearly identify source PR or branch
+    if [[ "$CIRRUS_BRANCH" != "master" ]]
+    then
+        NAME_SUFFIX="pr$CIRRUS_PR"
+    else
+        NAME_SUFFIX="$CIRRUS_BRANCH"
+    fi
+
+    # Same script re-used for both uploading and downloading to avoid duplication
```

Comment:
See symlink comment below; but much more importantly, would you consider changing the name(s)? It will be difficult for a maintenance programmer to keep track of which function is which depending on the order of `cache` and `release`; can you give one of them a more descriptive and unique name?

Reply:
Yes, happy to fix names.  I want to get this into some form of "kind of working" first though.  I'm not entirely sure the overall design here is sound or not.

Reply:
okay, renamed them.